src/v/raft/consensus.cc

```diff
@@ -1046,9 +1046,7 @@ void consensus::dispatch_vote(bool leadership_transfer) {
                           }
                           // background
                           ssx::spawn_with_gate(
-                            _bg,
-                            [vstm = std::move(vstm),
```

Comment:
not just dead, but i wonder why clang-tidy didn't warn against what looks like a use-after-move (or at least a move-after-move).

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/bugprone/use-after-move.html#use:
> An exception to this are objects of type std::unique_ptr, std::shared_ptr and std::weak_ptr, which have defined move behavior (objects of these classes are guaranteed to be empty after they have been moved from). Therefore, an object of these classes will only be considered to be used if it is dereferenced, i.e. if operator*, operator-> or operator[] (in the case of std::unique_ptr<T []>) is called on it.

Reply:
oh interesting TIL